qbft/spectest/tests/messages/valid_hash_data_root.go

```diff
@@ -1,8 +1,30 @@
 package messages
 
-import "github.com/bloxapp/ssv-spec/qbft/spectest/tests"
+import (
+	"github.com/bloxapp/ssv-spec/qbft"
+	"github.com/bloxapp/ssv-spec/qbft/spectest/tests"
+	"github.com/bloxapp/ssv-spec/types"
+	"github.com/bloxapp/ssv-spec/types/testingutils"
+)
 
 // ValidHashDataRoot tests a valid hash data root
 func ValidHashDataRoot() tests.SpecTest {
-	panic("implement")
+	ks := testingutils.Testing4SharesSet()
+
+	msg := testingutils.SignQBFTMsg(ks.Shares[1], types.OperatorID(1), &qbft.Message{
+		MsgType:    qbft.ProposalMsgType,
+		Height:     qbft.FirstHeight,
+		Round:      qbft.FirstRound,
+		Identifier: []byte{1, 2, 3, 4},
+		Root:       testingutils.TestingQBFTRootData,
+	})
+
+	msg.FullData = testingutils.TestingQBFTFullData
+
+	return &tests.MsgSpecTest{
+		Name: "valid hash data root",
+		Messages: []*qbft.SignedMessage{
+			msg,
+		},
+	}
 }
```

Comment:
I don't think this is what @alonmuroch meant to implement... but I am not sure what he did mean

Reply:
I'm also confused on this. :/